### PR TITLE
Add Token Gating

### PR DESCRIPTION
**Pull request by zfaizal2**
Allows one token with an amount to be added

Should only allow SPL token gate option if NFT option isn't selected, or isn't enabled on the forum already. Forum gating only enabled in "edit forum" and topic gating in "create topic" for SPL token. Enter mint address of a sample token and amount, and gating on a forum/topic should be enabled. 

image

### CODE REVIEW

**Comment by zfaizal2** [comment]
preview link https://forum-staging-edtopgy6u-usedispatch.vercel.app/

**Comment by ana2201** [comment]
The select for the token gate doesn't match the rest of the app
I suggest you add some padding, match the color of the select with some other color of the view (maybe a light purple like the border between rows) and add a border radius like the rest of the inputs in the modal have. Maybe also change the color when hovering

image
Menu_001

**Comment by ana2201** [comment]
Also, when multiple tokens are added style the list as it is on the `createForum` component

image

File: src/components/forums/forum/ForumContent.tsx [hunk 1]
```diff
@@ -346,20 +457,75 @@ export function ForumContent(props: ForumContentProps) {
...
+                              step={0.01}
+                              className="newAccessToken"
+                              onChange={(value) => {
+                                setNewForumAccessTokenAmount(parseFloat(value));
+                              }
+                              }/>
+                          </div>
+                            )}
+                  {newForumAccessToken.length > 0 && (
+                    accessList.map((pubkey) => (
```
Comment (ana2201) [inline src/components/forums/forum/ForumContent.tsx:524]: when multiple tokens are added, style the list as it is on the `createForum` component

image

File: src/utils/parseErrors.tsx [hunk 2]
```diff
@@ -29,9 +29,9 @@ const postboxErrorCode = {
 const hexToDecimal = (hex: string) => parseInt(hex, 16);
 
 export function parseError(error: any): DispatchError {
-  if (error.message != undefined) {
+  if (error?.message != undefined) {
```
Comment (ana2201) [inline src/utils/parseErrors.tsx:32]: use `!isNil(error?message)` instead

File: src/utils/parseErrors.tsx [hunk 3]
```diff
@@ -29,9 +29,9 @@ const postboxErrorCode = {
...
     const hexIndex = (error.message as string).indexOf("0x");
-    if (hexIndex >= 0) {
+    if (hexIndex >= 0 && error.message != undefined) {
```
Comment (ana2201) [inline src/utils/parseErrors.tsx:34]: same here ^

**Comment by zfaizal2** [comment]
i added some basic styling, we can improve styling in a future PR if necessary as well

**Pull request merged**